pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -108,3 +120,18 @@ func (p *Probe) checkServerInfo() error {
 
 	return nil
 }
+
+// pingListeners checks to ensure that Envoy is actually listenening on the port.
+func (p *Probe) pingInboundVirtualListener() error {
+
+	// Check if Inbound traffic capture ports is actually listening.
+	con, err := net.Dial("tcp", fmt.Sprintf("%s:%d", p.LocalHostAddr, networking.ProxyInboundListenPort))
```

Comment:
ohh.. this isnt good. We should externalize this setting in mesh config rather than relying on the hard coded value in pilot code. The moment it changes, older proxies would break. isnt there any way around this? is it very heavy to do a config dump and extract the orig dst listeners and ping them?

Reply:
Yeah. I was thinking about the same. Hitting `/listeners` is the alternative which I think is the best one - which I have implemented in my first commit, later changed to this just to verify. On a second thought, I will go back to `/listeners' endpoint approach which is not a heavy weight thing to do atleast on the startup and cache the result.

Reply:
Changed, PTAL.